glib/src/object.rs

```diff
@@ -1264,15 +1264,19 @@ pub trait ObjectExt: ObjectType {
     where
         N: Into<&'a str>,
         F: Fn(&[Value]) -> Option<Value>;
-    fn emit<'a, N: Into<&'a str>>(
+    /// Emit signal by signal id.
+    fn emit(&self, signal_id: SignalId, args: &[&dyn ToValue]) -> Result<Option<Value>, BoolError>;
+    fn emit_by_name<'a, N: Into<&'a str>>(
         &self,
         signal_name: N,
         args: &[&dyn ToValue],
     ) -> Result<Option<Value>, BoolError>;
-    fn emit_generic<'a, N: Into<&'a str>>(
```

Comment:
Why do you remove this one? And we should probably instead of removing this, also add generic variants for the two new functions :)

Also if you have suggestions for the name, something else than generic would be nice. `_with_values()`?

Reply:
Note that there is also other "generic" API, for setting properties for example. That should be changed consistently.

Reply:
> Why do you remove this one? And we should probably instead of removing this, also add generic variants for the two new functions :)

Ah, just noticed the difference between both emit & emit_generic. Maybe `emit_with_values`?

Reply:
And `emit_by_name_with_values()` and `emit_with_details_and_values()` and `set_property_with_values()`, etc?

Apart from the details one this seems like an improvement.